QModManager/Injector.cs

```diff
@@ -93,27 +91,36 @@ internal void Remove()
                     Environment.Exit(0);
                 }
 
-                if (File.Exists(backupFilename))
-                {
-                    File.Delete(mainFilename);
+                // Remove backup file if it exists
+                string backupFilePath = Path.Combine(managedDirectory, "Assembly-CSharp.qoriginal.dll");
+                if (File.Exists(backupFilePath))
+                    File.Delete(backupFilePath);
 
-                    File.Move(backupFilename, mainFilename);
+                AssemblyDefinition game = AssemblyDefinition.ReadAssembly(mainFilename);
 
-                    Console.WriteLine();
-                    Console.WriteLine("QModManager was uninstalled successfully");
-                    Console.WriteLine();
-                    Console.WriteLine("Press any key to exit...");
-                    Console.ReadKey();
-                    Environment.Exit(0);
+                TypeDefinition gameInputDef = game.MainModule.GetType("GameInput");
+                MethodDefinition awakeMethod = gameInputDef.Methods.First(x => x.Name == "Awake");
+
+                Instruction patchMethodCall = null;
+
+                foreach(Instruction instruction in awakeMethod.Body.Instructions)
+                {
+                    if(instruction.OpCode == OpCodes.Call && instruction.Operand.ToString().Equals("System.Void QModInstaller.QModPatcher::Patch()"))
+                    {
+                        patchMethodCall = instruction;
+                    }
                 }
 
-                Console.WriteLine();
-                Console.WriteLine("Cannot uninstall, file 'Assembly-CSharp-qoriginal.dll' is missing");
-                Console.WriteLine("To uninstall, you will need to verify game contents ons steam");
+                if(patchMethodCall != null)
+                    awakeMethod.Body.GetILProcessor().Remove(patchMethodCall);
```

Comment:
If its not null, you need to tell the user that the IsInjected check succeeded, but the patch method call wasn't found. (`An unexpected error has occurred.` and exit with exit code `3`)

Reply:
There is no way for that to happen considering the IsInjected check uses the same condition that I use to get the call instruction.

Reply:
So the null check wasn't even required, its just "there".

Reply:
Even if its not required, its better to have it